src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/WebJobsTelemetryInitializer.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    internal class WebJobsTelemetryInitializer : ITelemetryInitializer
+    {
+        private const string WebsiteInstanceIdKey = "WEBSITE_INSTANCE_ID";
+        private const string ComputerNameKey = "COMPUTERNAME";
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            telemetry.Context.Cloud.RoleInstance = GetRoleInstanceName();
```

Comment:
please cache this value in class's private field. If you need to re-read it periodically - cache it only for certain time